api/src/main/java/org/eclipse/microprofile/openapi/annotations/extensions/Extension.java

```diff
@@ -24,17 +24,62 @@
 import java.lang.annotation.Target;
 
 /**
- * A named extension that should be added to the OpenAPI definition.
+ * A named extension that should be added to the OpenAPI definition. The names of all extensions MUST begin with
+ * {@code x-} or else an invalid document will potentially be created.
+ * 
+ * <p>
+ * Although this annotation may currently be placed directly on a Java language element target, application developers
+ * should instead utilize the {@code extensions} property of the particular annotation that corresponds to the model
+ * being extended. Use of the annotation directly on a Java element is often ambiguous and it may result in the
+ * extension being added to an incorrect location in the OpenAPI model. Future releases of MicroProfile OpenAPI may
+ * remove the capability of placing this annotation directly on a Java element.
+ * 
+ * <p>
+ * When {@literal @}Extension annotations are used both directly on a Java element as well as within another annotation
```

Comment:
This whole javadoc is really good. :)

I think usually I expect to see class and annotation names in a `{@code }` block (as you've done for the `extensions` property). Maybe that's a matter of style.

Reply:
I prefer `{@code }` as well - this had been updated.